tests/system/Helpers/InflectorHelperTest.php

```diff
@@ -285,4 +285,41 @@ public function testOrdinalize()
             $this->assertSame($suffixed, $ordinalized);
         }
     }
+
+    public function testDecamelizeToSnakeCase()
+    {
+        $strings = [
+            'simpleTest'      => 'simple_test',
+            'easy'            => 'easy',
+            'HTML'            => 'html',
+            'simpleXML'       => 'simple_xml',
+            'PDFLoad'         => 'pdf_load',
+            'startMIDDLELast' => 'start_middle_last',
+            'AString'         => 'a_string',
+            'Some4Numbers234' => 'some4_numbers234',
+            'TEST123String'   => 'test123_string',
+            'hello_world'     => 'hello_world',
+            'hello___world'   => 'hello___world',
+            '_hello_world_'   => '_hello_world_',
+            'HelloWorld'      => 'hello_world',
+            'helloWorldFoo'   => 'hello_world_foo',
+            'hello_World'     => 'hello_world',
+            'hello-world'     => 'hello-world',
+            'myHTMLFiLe'      => 'my_html_fi_le',
+            'aBaBaB'          => 'a_ba_ba_b',
+            'BaBaBa'          => 'ba_ba_ba',
+            'libC'            => 'lib_c',
+            'a'               => 'a',
+            'A'               => 'a',
+            'aB'              => 'a_b',
+            'AB'              => 'ab',
```

Comment:
This variance bothers me a little bit, but ultimately I don't know that two characters are enough to be meaningful anyways, so if you're okay with these then I won't make a fuss.

Reply:
I'm not concerned with this, personally. It follows the same logic as for longer strings, and I would expect the consistency, not special treatment for shorter strings. I'll merge now and we can always revisit if we need to later.